README.md

```diff
@@ -57,95 +57,388 @@ Some of the skills/concepts this assignment aims to test (and build upon):
 * Routing (URL fragment based routing)
 
 ## API
-We can build API documentation using [Swagger](https://swagger.io/) or API Blueprint. 
+We can build API documentation using [Swagger](https://swagger.io/) or API Blueprint.
 A simple express JS API server can be built that is provided to students to allow them to develop offline. It could also be hosted somewhere to ease development when online?
 
 TODO: Someone will need to build a API server to support all of this client side functionality.
 
 The following specification is a WIP and only includes some endpoints.
 
+> POST `/login`
+
+**Request**
 ```json
-POST /login
 {
   "username": "string",
   "password": "string",
 }
-returns:
+```
+
+**Response**
+`200` On successful login, sets a cookie with a session token
+```json
+{
+  "error": false,
+  "errorMsg": null
+}
+```
 
-200 OK for successful login
-403 for invalid credentials
-POST /signup
+`403` for invalid credentials
+```json
 {
-  "displayName": "string",
+  "error": true,
+  "errorMsg": "Invalid Credentials"
+}  
+```
+
+> POST `/signup`
+
+**Request**
+```json
+{
+  "email": "string",
   "username": "string",
   "password": "string",
 }
-returns
+```
+**Response**
+`200` On successful signup, sets a cookie with a session token
+```json
+{
+  "error": false,
+  "errorMsg": null
+}
+```
+
+`403` for existent username
+```json
+{
+  "error": true,
+  "errorMsg": "Username Already In Use"
+}  
+```
+
+> PUT `/post/like?postId=13`
+
+**Request**
+```
+postId : Id of the post the current logged in user wishes to like
+```
+
+**Response**
+`200` On successful like
+```json
+{
+  "error": false,
+  "errorMsg": null
+}
+```
+
+`403` for Non Existent post or user not logged in
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Post Id"
+}  
+```
+
+
+> PUT `/post/comment`
```

Comment:
Should be POST. "creating an entry". Maybe to standardise any id should be url param?